src/libraries/System.Data.OleDb/src/UnsafeNativeMethods.COMWrappers.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Data.Common
+{
+    internal static partial class UnsafeNativeMethods
+    {
+        //
+        // Oleaut32
+        //
+
+        [DllImport(Interop.Libraries.OleAut32, CharSet = CharSet.Unicode, PreserveSig = true)]
+        internal static extern System.Data.OleDb.OleDbHResult GetErrorInfo(
+            [In] int dwReserved,
+            [Out] out System.IntPtr ppIErrorInfo);
```

Comment:
Don't use `out` in P/Invokes, but instead use pointers. This matches the native signatures as close as possible, and allows for the native marshalling to do less work. In this case, the native marshalling needs to pin the managed reference `out IntPtr ppIErrorInfo`. But it doesn't if the signature is a pointer.

```suggestion
            [Out] System.IntPtr* ppIErrorInfo);
```

Reply:
Please also drop the `[Out]` attribute.